README.md

```diff
@@ -39,54 +39,38 @@ ant test
 
 See the [documentation for api usage].
 
-After a user completes the oauth2 flow, the credentials for their account will be stored as a data source with a configured folder created in the user's root mailbox during authentication. If an import class is associated, it can be triggered manually with a `zmsoap` import data request.
+After a user completes the oauth2 flow, the credentials for their account will be stored as a data source with a configured folder created in the user's root mailbox during authentication. An import can be triggered manually with a `zmsoap` import data request.
 
 ---
 
 ## Configuration
 
-This service's configuration can all be found in Zimbra's `localconfig.xml` file (usually located at `/opt/zimbra/conf/localconfig.xml`)
-
-**Localconfig General Properties**
+This service's configuration for OAuth clients are setup in Ldap attributes on a global or domain level.
 
+**Ldap Properties**
 
-| Key | Description | Optional | Example Options |
-| --- | ----------- | -------- | --------------- |
-| host_uri_template | The host uri to connect via ZMailbox | Yes | `https://%s:443` |
-| zm_oauth_classes_handlers_yahoo<sup>1</sup> | The handler implementation class for the client | | `com.zimbra.oauth.handlers.impl.YahooOAuth2Handler` |
+See the [client setup wiki].
 
-<sup>1</sup>Replace the `yahoo` part of the key name with the name of the client (e.g. `yahoo`, `google`, `outlook`).
+| Key | Description | Required By | Template |
+| --- | ----------- | ----------- | -------- |
+| zimbraOAuthConsumerCredentials | OAuth credentials for a client, set at global config or domain level. | All | `<client-id>:<client-secret>:<client>` |
+| zimbraOAuthConsumerRedirectUri | The callback where the client returns the user too. | All | `http[s]://<domain[:port]>/service/extension/oauth2/authenticate/<client>:<client>` |
+| zimbraOAuthConsumerAPIScope | The scopes required to access user data. | Google, Facebook | `<scope1>,<scope2>,...:<client>` |
 
+***Client specific scopes to use with the zimbraOAuthConsumerAPIScope config***
 
-**Localconfig Client Specific Properties**
-
-**Yahoo Implementation Properties**
-
-| Key | Description | Optional | Example Options |
-| --- | ----------- | -------- | --------------- |
-| zm_oauth_yahoo_client_id | The Yahoo app's client id | | |
-| zm_oauth_yahoo_client_secret | The Yahoo app's client secret | | |
-| zm_oauth_yahoo_client_redirect_uri | The callback Yahoo returns the user to | | `https://this.service.host.com/service/extension/oauth2/authenticate/yahoo` |
+| Client | Required scopes string |
+| ------ | ---------------------- |
+| Google | `https://www.googleapis.com/auth/contacts profile:google` |
+| Facebook | `user_friends,read_custom_friendlists,email,user_location,public_profile,user_about_me,user_birthday,groups_access_member_info:facebook` |
```

Comment:
Are all of these necessary for fb to work? I would either list ONLY the ones we use, or just add the primary ones to the required_scopes constant and list excess here. This can be done for Google as well.